symbolic-symcache/src/new/writer.rs

```diff
@@ -308,6 +313,113 @@ impl SymCacheConverter {
         }
     }
 
+    #[cfg(feature = "il2cpp")]
+    /// Processes a set of [`UsymSymbols`], passing all mapped symbols into the converter.
+    pub fn process_usym(&mut self, usym: &UsymSymbols) -> Result<(), SymCacheError> {
+        // Assume records they are sorted by address; There's a test that guarantees this
+        let mapped_records = usym.records().filter_map(|r| match r {
+            UsymSourceRecord::Unmapped(_) => None,
+            UsymSourceRecord::Mapped(r) => Some(r),
+        });
+
+        let mut curr_id: String = String::default();
+        let mut function_idx = 0;
+        for record in mapped_records {
+            // like process_symbolic_function, skip functions whose address is too large to fit in a
+            // u32
+            if record.address > u32::MAX as u64 {
+                continue;
+            }
+            let address = record.address as u32;
+
+            // Records that belong to the same function will have the same identifier.
+            // Symbols have GUID-like sections to them that might ensure they're unique across
+            // files, but we'll just include the file name and paths to be very safe.
+            let identifier = format!("{}::{}", record.native_file, record.native_symbol);
```

Comment:
I'd have probably made a custom struct for this with two borrowed references, now you're allocating and formatting strings for each of these.

Reply:
it's a bit difficult to get rid of the allocations, but the formatting i can do something about. i've pushed a change that takes care of that at least, so please take a look when you're back 🙏